### ISSUE TITLE
Hard to install/package

### ISSUE THREAD

**Issue by poelzi** [actboy168/luamake#31]
Tried it shortly to install, but so far, this is hard/not packageable whichout headache.
luamake should search a compile time defined path for lua scripts it ships. Those need to be installed into a absolute target, like 
/usr/local/[lib|share] and the file installed into $PREFIX/bin needs to search this path.
Changing $HOME/.profile|.... is a strict nogo for every unix package.

**Comment by nospam2998** [comment]
Could we please have another take on this, and be a bit specific about what the actual problems are?

**TL;DR: Would you be open for merging a change to [publish.yml][], one creating source releases?**

Sure, a fragmented world might be a challenging reality. Yet as you write @actboy168,  package management systems exist to deal with it. Shortly prior to this issue, there was a request in #29 for introducing versioned releases. A request which has so far only seen a `enhancement` flag and no comments.

One need not look hard at the repository to realize there is a workflow called [publish.yml][], which however to the best of my understanding does not appear to actually publish anything?

How would you welcome a few very small but concrete steps towards improving the situation:

 1. Rename `publish.yml` to `build.yml`, to better have it reflect the current situation.
 2. Create a new `publish.yml` with a job named `source`. A job which watches for new tags; initiating git modules, building a usable tar ball and storing it as a github release when it happens.
 3. Release a version once appropriate to do so, and then automatically get a source release from the above suggested trivial workflow.

Step 1 is **not** important. Adding a source job to the existing workflow would also work just fine.

The point is **not** that any competent person has trouble using the git distributed install script to run things on their own machine. What **is hard** is to do it in the reproducible/structured manner required to include software for package distributions in accordance with their processes.

To be concrete, the git-yolo distribution format of luamake and bee.lua have seemingly stopped all progress on [bringing LuaLS to luarocks][luarocks], as early as during the idea phase. As late as last week I rediscovered this, in an experimental attempt at bringing LuaLS to [freebsd-ports][].

[luarocks]: https://github.com/LuaLS/lua-language-server/discussions/623
[publish.yml]: https://github.com/actboy168/luamake/blob/f41f6ef/.github/workflows/publish.yml
[freebsd-ports]: https://github.com/LuaLS/lua-language-server/issues/2361#issuecomment-2184069525

**Comment by actboy168** [comment]
@nospam2998 You can submit a pull request to improve this.
